### DISCUSSION TITLE
Clarification on Support for Multiple Kubernetes Versions

### DISCUSSION THREAD

**Discussion by aleixam-klarrio** [vmware-tanzu/pinniped#2060] [Q&A]
Hi Pinniped team,

I have recently been going through the documentation and am looking for clarification on a specific aspect. I am trying to determine if the latest version of Pinniped fully supports multiple Kubernetes versions.

While I see that there is auto-generated code to help with deployment across different Kubernetes versions, the documentation still isn't very clear to me on the compatibility details.

Could you please provide more information on:

- Which Kubernetes versions are officially supported by the latest version of Pinniped?
- Are there any known limitations or compatibility issues when using Pinniped with older versions of Kubernetes?
- Does Pinniped dynamically adapt to different API versions based on the Kubernetes cluster it's running on, or are there specific considerations we need to make during deployment?

Thanks in advance for your help and for maintaining this great project!

**Comment by cfryanr** [comment]
Hi, while we do not officially publish a compatibility matrix, we do strive to support a wide number of Kubernetes versions. Each time we create a new release, it has already been tested with whatever is the latest version of Kubernetes at that time, along with many previous versions. Generally, it should "just work". For example, the most recent release was tested every minor version of Kubernetes from 1.21 through 1.31. If you ever have any trouble, please let's know!
